bundles/mapping/mapmodule/service/HoverHandler.js

```diff
@@ -0,0 +1,265 @@
+            this.state.renderFeatureId = feature.get(FTR_PROPERTY_ID);
+            layer.changed();
+        } else {
+            layer.getSource().addFeature(feature);
+        }
+    }
+
+    onMapHover (event) {
+        this._updateTooltipPosition(event);
+    }
+
+    createHoverLayer (layer, source) {
+        const olLayer = this._getLayer(layer, source);
+        olLayer.setOpacity(layer.getOpacity());
+        olLayer.setVisible(layer.isVisible());
+        olLayer.set(LAYER_HOVER, true, true);
+        olLayer.setStyle(this._styleGenerator(layer));
+        this.olLayers[layer.getId()] = olLayer;
+        this.setTooltipContent(layer);
+        return olLayer;
+    }
+
+    _getLayer (layer, source) {
+        if (layer.getLayerType() === VECTOR_TILE_TYPE) {
+            return new olLayerVectorTile({ source });
+        }
+        return new olLayerVector({ source: new olSourceVector() });
+    }
+
+    setTooltipContent (layer) {
+        const options = layer.getHoverOptions();
+        if (!options || !Array.isArray(options.content)) {
+            return;
+        }
+        this._tooltipContents[layer.getId()] = options.content;
+    }
+
+    setDefaultStyle (layerType, styleType, styleDef) {
+        const styles = this._defaultStyles[layerType] || {};
+        styles[styleType] = styleDef;
+        this._defaultStyles[layerType] = styles;
+    }
+
+    _styleGenerator (layer) {
+        const { featureStyle: layerHoverDef } = layer.getHoverOptions() || {};
+        const { featureStyle: defaultFeatureStyle, hover: defaultHoverDef } = this._defaultStyles[layer.getLayerType()] || {};
+        let hoverDef = layerHoverDef || defaultHoverDef;
+        if (hoverDef) {
+            if (hoverDef.inherit === true) {
+                let layerStyleDef = layer.getCurrentStyleDef() || {};
+                if (!layerStyleDef.featureStyle) {
+                    // Bypass possible layer definitions
+                    Object.values(layerStyleDef).find(obj => {
+                        if (obj.hasOwnProperty('featureStyle')) {
+                            layerStyleDef = obj;
+                            return true;
+                        }
+                    });
+                }
+                hoverDef = jQuery.extend(true, {}, defaultFeatureStyle, layerStyleDef.featureStyle, hoverDef);
+            }
+            // TODO: if layer contains only one geometry type return olStyle (hoverDef) instead of function
+            const olStyles = this.styleFactory(hoverDef);
+            if (layer.getLayerType() === VECTOR_TILE_TYPE) {
+                return feature => {
+                    if (this.state.renderFeatureId === feature.get(FTR_PROPERTY_ID)) {
+                        return getStyleForGeometry(feature.getType(), olStyles);
+                    }
+                };
+            }
+            return feature => {
+                return getStyleForGeometry(feature.getGeometry(), olStyles);
+            };
+        }
```

Comment:
Reverse this so don't get as much indentation:
```suggestion
        if (!hoverDef) {
            return null;
        }
        if (hoverDef.inherit === true) {
            let layerStyleDef = layer.getCurrentStyleDef() || {};
            if (!layerStyleDef.featureStyle) {
                // Bypass possible layer definitions
                Object.values(layerStyleDef).find(obj => {
                    if (obj.hasOwnProperty('featureStyle')) {
                        layerStyleDef = obj;
                        return true;
                    }
                });
            }
            hoverDef = jQuery.extend(true, {}, defaultFeatureStyle, layerStyleDef.featureStyle, hoverDef);
        }
        // TODO: if layer contains only one geometry type return olStyle (hoverDef) instead of function
        const olStyles = this.styleFactory(hoverDef);
        if (layer.getLayerType() === VECTOR_TILE_TYPE) {
            return feature => {
                if (this.state.renderFeatureId === feature.get(FTR_PROPERTY_ID)) {
                    return getStyleForGeometry(feature.getType(), olStyles);
                }
            };
        }
        return feature => {
            return getStyleForGeometry(feature.getGeometry(), olStyles);
        };
```